lib/jsonapi/authorization/authorizing_processor.rb

```diff
@@ -181,27 +186,32 @@ def authorize_replace_to_many_relationship
         )
       end
 
-      def authorize_remove_to_many_relationship
+      def authorize_remove_to_many_relationships
         source_resource = @resource_klass.find_by_key(
           params[:resource_id],
           context: context
         )
         source_record = source_resource._model
 
-        relationship_type = params[:relationship_type].to_sym
-        related_resource = @resource_klass
-          ._relationship(relationship_type)
+        related_resources = @resource_klass
+          ._relationship(params[:relationship_type].to_sym)
           .resource_klass
-          .find_by_key(
-            params[:associated_key],
+          .find_by_keys(
+            params[:associated_keys],
             context: context
           )
-        related_record = related_resource._model unless related_resource.nil?
+
+        # find_by_keys doesn't raise if no records, so raise explicitly here
+        if related_resources.empty?
+          fail JSONAPI::Exceptions::RecordNotFound, params[:associated_keys]
```

Comment:
Good question! If this is removed, one [spec](https://github.com/handlers/jsonapi-authorization/blob/jsonapi-resources-0.9/spec/requests/relationship_operations_spec.rb#L302) failure occurs: when "authorized for remove_to_many_relationship" and "limited by policy scope on comments" it `{ is_expected.to be_not_found }`. The spec expects a 404 if the comments scope (`CommentPolicy::Scope`) is `Comment.none`.

With the new code, it instead tries to run `remove_from_comments?`, which raises `NotImplementedError`.

```ruby
  1) Relationship operations DELETE /articles/:id/relationships/comments authorized for remove_to_many_relationship limited by policy scope on comments
     Failure/Error: raise NotImplementedError

     NotImplementedError:
       NotImplementedError
     # ./spec/dummy/app/policies/article_policy.rb:44:in `remove_from_comments?'
     # ./lib/jsonapi/authorization/default_pundit_authorizer.rb:238:in `public_send'
     # ./lib/jsonapi/authorization/default_pundit_authorizer.rb:238:in `authorize_relationship_operation'
     # ./lib/jsonapi/authorization/default_pundit_authorizer.rb:183:in `remove_to_many_relationship'
     # ./lib/jsonapi/authorization/authorizing_processor.rb:201:in `authorize_remove_to_many_relationships'
     # ./spec/requests/relationship_operations_spec.rb:275:in `block (3 levels) in <top (required)>'
     # ./spec/requests/relationship_operations_spec.rb:302:in `block (5 levels) in <top (required)>'
```

I included the `fail` to preserve the old 404 behavior. Perhaps we could move that `fail` to `authorize_relationship_operation`?